doc/compiler/CompilerOptions.md

```diff
@@ -0,0 +1,94 @@
+<!--
+Copyright (c) 2016, 2018 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath 
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# OMR Compiler Options
+
+The OMR Compiler Technology provides a number of options that allow you to take
+some control over its run-time behavior. This document contains the following:
+* [How compiler options can be set](#setting-options)
+* [Option Sets](#option-sets)
+
+To see all of the available options to control the compiler, please see the options
+listed in [`<omr-root-dir>/compiler/control/OMROptions.cpp`](https://github.com/eclipse/omr/blob/e2f65411e67d21ef04e2062a8945e604d82cb19e/compiler/control/OMROptions.cpp#L98)
+and [`<omr-root-dir>/compiler/env/FEBase.cpp`](https://github.com/eclipse/omr/blob/e2f65411e67d21ef04e2062a8945e604d82cb19e/compiler/env/FEBase.cpp#L235)
+
+# Setting options
+
+OMR Compiler options are either set through command line options, or environment
+variables. If OMR Compiler is being used as part of a language VM, then the
+options can only be set through the VM command line arguments using the `-Xjit`
+argument followed by the options, or through environment variables in some cases,
+such as `TR_dumpGraphs`.
+
+For language-independent applications of the OMR Compiler Technology such as
+the Test Compiler, the compiler options can be set using the environment variable
+`TR_Options`:
+
+```
+TR_Options='traceFull,log=log' ./testjit
+```
+
+Below is an example of how the same options can be set in the J9 JVM.
+
+```sh
+java -Xjit:count=0,traceFull,log=log HelloWorld
+```
+
+
+# Option Sets
+
+It is possible to apply a set of compiler options ("option sets") only to a
+method, or group of methods. Furthermore, it is possible to apply different
+option sets to different (groups of) methods. An option set can be defined on
+the command line by enclosing the desired set of parameters in parentheses,
+and preceding it with a method filter expression, e.g.
```

Comment:
...and preceding it with a method filter expression in braces.  For example, to apply the option set `count=0,optlevel=hot` only to the method that matches the signature `HelloWorld.main([Ljava/lang/String;)V` add the following to the `TR_Options` environment variable or `-Xjit` command-line: